### DISCUSSION TITLE
[BP#58] Générer les PDF en dehors du CMS

### DISCUSSION THREAD

**Discussion by devcrafting** [cnumr/best-practices#213] [Bonnes pratiques] [answered]
J'ai beaucoup de mal à voir à quoi il est fait référence dans cette BP 🤔
Elle semble super spécifique aux CMSet peut-être même à un CMS qui aurait une fonctionnalité particulière qui fait beaucoup de PDF à la volée ??

**Comment by ACTLEM** [comment]
Oui, il existe des extensions de CMS (comme pour Wordpress) pour générer des PDF.

**Reply by devcrafting** [reply]
peut-être juste mieux contextualiser (à vrai dire les BP autour des CMS m'ont laissé perplexe à la base ^^)

**Reply by arnolem** [reply]
A une époque lointaine, Mambo, qui est devenu Joomla, proposait un bouton pour afficher les pages en PDF.
Il générait les pages en PDF à la volée et je sais que c'était une fonctionnalité demandé par des collectivités par exemple.
Si la BP parle de ça, je ne sais pas si c'est vraiment d'actualité car ça me semble des pratiques anecdotiques aujourd'hui.
En 2021, on trouve plutôt des boutons "imprimer" qui lance l'impression navigateur.

A la lecture de la BP, je me disais que l'on parlait plutôt des PDF généré à la volé dans des applications web comme un extranet avec des factures clients uploadées plutôt que générée à la volée.

**Reply by florinesueur** [reply]
Je me demande si il ne faut pas généraliser davantage comme : Optimiser et générer vos médias hors navigateur.

Exemple : 
- Recouper son image dans wordpress
- Générer les pdf
- Générer les thumbnails
- Optimisser ses images avec un plugin ou sharp pour un site statique (qui tue le temps de build à chaque fois :))
- Idem pour les vidéos et l'audio

**Comment by rlemaire** [comment]
L'idée est d'optimiser les contenus (images, pdf, audio, vidéo, ...à avant leur upload, dans un CMS ou équivalent. Il y a plusieurs pratiques qui vont dans ce sens.
On part du principe que ce n'est pas fait automatiquement mais que c'est à la charge des gens qui gèrent le contenu en question, plutôt peu techniques.

**Comment by ACTLEM** [comment]
Donc trois possibilités:
- On laisse la BP telle quelle
- On la supprime
- On la modifie pour élargir aux différents "gros" contenus (cf remarque de @florinesueur @rlemaire )

Votre avis?

**Reply by rlemaire** [reply]
Je vote pour la troisième option.

**Reply by davidgroult** [reply]
3ème option également.

**Comment by ACTLEM** [comment, marked as answer]
J'ai créé l'issue pour élargir la BP à plusieurs contenus: https://github.com/cnumr/best-practices/issues/304

@florinesueur je me suis permis de te l'affecter car tu étais à l'initiative de cette proposition
